library/ssl_msg.c

```diff
@@ -1135,32 +1147,38 @@ int mbedtls_ssl_decrypt_buf( mbedtls_ssl_context const *ssl,
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
 
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
-        status = psa_cipher_decrypt_setup( &cipher_op,
-                                     transform->psa_key_dec, transform->psa_alg );
+        /* Skip psa decryption for null cipher */
+        if ( transform->psa_alg != MBEDTLS_SSL_NULL_CIPHER )
```

Comment:
Same as encrypt: this is the stream case, which unless I'm deeply mistaken is now equivalent to NULL, so we can probably remove the non-NULL code and just make it return an error instead.

Reply:
To be done when CI passes.

Reply:
The CI's passing, so I think now is the time :)